lib/ansible/cli/galaxy.py

```diff
@@ -1591,36 +1592,55 @@ def execute_search(self):
         page_size = 1000
         search = None
 
+        # role or collection
+        content = context.CLIARGS['type']
+
         if context.CLIARGS['args']:
             search = '+'.join(context.CLIARGS['args'])
 
         if not search and not context.CLIARGS['platforms'] and not context.CLIARGS['galaxy_tags'] and not context.CLIARGS['author']:
             raise AnsibleError("Invalid query. At least one search term, platform, galaxy tag or author must be provided.")
 
-        response = self.api.search_roles(search, platforms=context.CLIARGS['platforms'],
-                                         tags=context.CLIARGS['galaxy_tags'], author=context.CLIARGS['author'], page_size=page_size)
+        if content == 'role':
+            response = self.api.search_roles(search, platforms=context.CLIARGS['platforms'], tags=context.CLIARGS['galaxy_tags'],
+                                             author=context.CLIARGS['author'], page_size=page_size)
+        elif content == 'collection':
+            response = self.api.search_collections(search, page_size=page_size)
+            # platforms=context.CLIARGS['platforms'], tags=context.CLIARGS['galaxy_tags'], author=context.CLIARGS['author'])
 
-        if response['count'] == 0:
-            display.display("No roles match your search.", color=C.COLOR_ERROR)
+        count = response.get('count', response.get('collection').get('count'))
+        if count == 0:
+            display.display("No %s match your search." % content, color=C.COLOR_ERROR)
             return True
 
         data = [u'']
 
-        if response['count'] > page_size:
-            data.append(u"Found %d roles matching your search. Showing first %s." % (response['count'], page_size))
+        if count > page_size:
+            data.append(u"Found %d %s matching your search. Showing first %s." % (count, content, page_size))
         else:
-            data.append(u"Found %d roles matching your search:" % response['count'])
+            data.append(u"Found %d %s matching your search:" % (count, content))
 
         max_len = []
-        for role in response['results']:
-            max_len.append(len(role['username'] + '.' + role['name']))
+        results = response.get('results', response.get('collection').get('results'))
+        for package in results:
+            if content == 'role':
+                max_len.append(len(package['username'] + '.' + package['name']))
+            else:
+                max_len.append(len(package['name']))
```

Comment:
`package['name']` doesn't include the namespace.
```suggestion
                max_len.append(len(package['namespace']['name'] + '.' + package['name']))
```

Reply:
it does, at least in the api i'm hitting

Reply:
I'm using the API at https://galaxy.ansible.com

With my change:
```
 Name                        latest
 ----                        ------
 newswangerd.collection_demo 1.0.11
```
Without:
```
 Name            latest
 ----            ------
 newswangerd.collection_demo 1.0.11
```

Reply:
ah, thought you meant getting the name, but you meant the padding (going to rework anyways to try to make better)